tests/publiccloud/create_aistack_env.pm

```diff
@@ -0,0 +1,247 @@
+    script_run("sh ./install_helm.sh");
+    script_run("helm version");
+    script_run("curl -sSL $kubectl_url -o ./kubectl && chmod +x ./kubectl");
+    script_run("sudo mv ./kubectl /usr/local/bin/");
+    script_run("kubectl version --client");
+}
+
+sub install_dependency_components {
+    my ($instance) = @_;
+    my $cert_repo = get_var('HELM_CERTS');
+    my $ingress_repo = get_var('HELM_INGRESS');
+    my $ing_ver = get_var('ING_VERSION');
+    # Add Ingress Controller to open-webui endpoint
+    assert_script_run("helm repo add $ingress_repo");
+    assert_script_run("helm repo update");
+    assert_script_run("helm upgrade --install ingress-nginx ingress-nginx/ingress-nginx --namespace ingress-nginx --set controller.service.type=ClusterIP --version $ing_ver --create-namespace", timeout => 120);
+
+    # Add cert-manager repo,install
+    assert_script_run("helm repo add $cert_repo");
+    assert_script_run("helm repo update");
+    assert_script_run("helm install cert-manager jetstack/cert-manager --namespace cert-manager --create-namespace --version v1.15.2 --set crds.enabled=true", timeout => 120);
+    sleep 180;
+}
+
+sub config_kubectl {
+    my ($instance) = @_;
+    # config kubectl
+    assert_script_run("mkdir -p ~/.kube");
+    assert_script_run("sudo cp /etc/rancher/rke2/rke2.yaml ~/.kube/config");
+    assert_script_run("kubectl config get-contexts");
+    assert_script_run("kubectl config use-context default");
+    assert_script_run("kubectl config view");
+}
+
+sub install_aistack_chart {
+    my ($instance, $ai_chart_repo, $namespace, $vf_name) = @_;
+    my $SECRET_application_collection = get_var('_SECRET_DOCKER');
+    my $cert_repo = get_var('HELM_CERTS');
+    my $ingress_repo = get_var('HELM_INGRESS');
+    my $docker_user_name = 'yarunachalam@suse.com';
+    #my $repo_url = 'github.com/SUSE/private-ai-charts';
+    my $repo_url = get_var('HELM_CHARTS');
+    my $git_token = get_var('_SECRET_SSH');
+    my $ing_ver = get_var('ING_VERSION');
+    #my $ing_ver = '4.11.1';
+
+    record_info('AISTACK charts install');
+    assert_script_run("helm list --all-namespaces");
+    assert_script_run("kubectl get pods --all-namespaces");
+
+    # Access to Application collection registery
+    # Get docker username and password
+    assert_script_run("kubectl create ns $namespace");
+    assert_script_run("kubectl create secret docker-registry application-collection --docker-server=dp.apps.rancher.io --docker-username='$docker_user_name' --docker-password='$SECRET_application_collection' -n $namespace", timeout => 120);
+
+    # Install private-ai-stack
+    my $gitlab_clone_url = 'https://git:' . $git_token . '@' . $repo_url;
+    assert_script_run("git clone $gitlab_clone_url");
+    assert_script_run("curl -o $vf_name $ai_chart_repo", timeout => 120);
+    assert_script_run("kubectl apply -f https://raw.githubusercontent.com/rancher/local-path-provisioner/v0.0.28/deploy/local-path-storage.yaml", timeout => 120);
```

Comment:
Please add this to the data directory instead of a direct download from internet

```suggestion
    assert_script_run("kubectl apply -f https://raw.githubusercontent.com/rancher/local-path-provisioner/v0.0.28/deploy/local-path-storage.yaml", timeout => 120);
```

See https://progress.opensuse.org/issues/169138 and https://progress.opensuse.org/issues/168448 for reasoning

Reply:
fixed